explainers/signed-exchange-subresource-subtitution-explainer.md

```diff
@@ -0,0 +1,455 @@
+      variant-key-05="image/jpeg";
+      header-integrity="sha256-AAA"
+Link: <https://publisher.example/image>;
+      rel="allowed-alt-sxg";
+      variants-05="accept;image/jpeg;image/webp";
+      variant-key-05="image/webp";
+      header-integrity="sha256-BBB"
+Link: <https://publisher.example/image>; rel=preload; as=image;
+```
+If a UA supports WebP format, the UA can use the signed exchange which
+header-integrity is sha256-BBB if available in the cache. Otherwise the UA can
+use the signed exchange which header-integrity is sha256-AAA if available in the
+cache.
+
+When the sha256-AAA signed exchange exists in the cache but the sha256-BBB
+signed exchange doesn’t exists, the UA which supports WebP format MUST ignore
+the sha256-AAA signed exchange and fetch the original URL. Otherwise this can be
+used for sending tracking ID.
+
+The **alternate** link also can have
+[variants and variant-key](https://httpwg.org/http-extensions/draft-ietf-httpbis-variants.html)
+attributes to support content negotiation for recursive prefetch.
+
+- In **outer** HTTP response of article.html.sxg from feed.example:
+    ```
+    Link: <https://feed.example/publisher.example/image_jpeg.sxg>;
+          rel="alternate";
+          type="application/signed-exchange;v=b3";
+          variants-05="accept;image/jpeg;image/webp";
+          variant-key-05="image/jpeg";
+          anchor="https://publisher.example/image";
+    Link: <https://feed.example/publisher.example/image_webp.sxg>;
+          rel="alternate";
+          type="application/signed-exchange;v=b3";
+          variants-05="accept;image/jpeg;image/webp";
+          variant-key-05="image/webp";
+          anchor="https://publisher.example/image";
+    ```
+- In **inner** response header of article.html.sxg:
+    ```
+    Link: <https://publisher.example/image>;
+          rel="allowed-alt-sxg";
+          variants-05="accept;image/jpeg;image/webp";
+          variant-key-05="image/jpeg";
+          header-integrity="sha256-AAA"
+    Link: <https://publisher.example/image>;
+          rel="allowed-alt-sxg";
+          variants-05="accept;image/jpeg;image/webp";
+          variant-key-05="image/webp";
+          header-integrity="sha256-BBB"
+    Link: <https://publisher.example/image>; rel=preload; as=image;
+    ```
+If a UA supports WebP, the UA should prefetch **image_webp.sxg** which content
+is WebP format. Otherwise the UA should prefetch **image_jpeg.sxg** which
+content is JPEG format.
+
+## Security and Privacy Considerations
+- The publishers can know whether the referrer page has prefetched the signed
+exchange subresources or not by checking the resource timing information. But
+this only exposes 1 bit information (= yes or no) because UAs can use the cached
+signed exchange only if the required signed exchanges are all available.
+- The UA can use the prefetched signed exchange subresources, only when they
+were prefetched in the referrer page. This is intended to avoid leaking the
```

Comment:
```suggestion
were prefetched in the immediate referrer page. This is intended to avoid leaking the
```

It's not clear to me exactly what information gets leaked if we do allow a series of navigations, especially same-origin navigations, to continue using the same set of prefetches. Maye the benefit isn't there, since the first page can just re-prefetch the same-origin content, but then we should say that instead of being vague about the leak.